java/client/src/main/java/glide/api/commands/Transaction.java

```diff
@@ -0,0 +1,149 @@
+            createOrAddCommand(Command.ping(msg));
+            return this;
+        }
+
+        /**
+         * Get information and statistics about the Redis server. DEFAULT option is assumed
+         *
+         * @see <a href="https://redis.io/commands/info/">redis.io</a> for details.
+         * @return CompletableFuture with the response
+         */
+        public TransactionBuilder info() {
+            createOrAddCommand(Command.info());
+            return this;
+        }
+
+        /**
+         * Get information and statistics about the Redis server.
+         *
+         * @see <a href="https://redis.io/commands/info/">redis.io</a> for details.
+         * @param options - A list of InfoSection values specifying which sections of information to
+         *     retrieve. When no parameter is provided, the default option is assumed.
+         * @return CompletableFuture with the response
+         */
+        public TransactionBuilder info(InfoOptions options) {
+            createOrAddCommand(Command.info(options.toInfoOptions()));
+            return this;
+        }
+
+        /**
+         * Get the value associated with the given key, or null if no such value exists.
+         *
+         * @see <a href="https://redis.io/commands/get/">redis.io</a> for details.
+         * @param key - The key to retrieve from the database.
+         * @return If `key` exists, returns the value of `key` as a string. Otherwise, return null
+         */
+        public TransactionBuilder get(String key) {
+            createOrAddCommand(Command.get(key));
+            return this;
+        }
+
+        /**
+         * Set the given key with the given value.
+         *
+         * @see <a href="https://redis.io/commands/set/">redis.io</a> for details.
+         * @param key - The key to store.
+         * @param value - The value to store with the given key.
+         * @return null
+         */
+        public TransactionBuilder set(String key, String value) {
+            createOrAddCommand(Command.set(key, value));
+            return this;
+        }
+
+        /**
+         * Set the given key with the given value. Return value is dependent on the passed options.
+         *
+         * @see <a href="https://redis.io/commands/set/">redis.io</a> for details.
+         * @param key - The key to store.
+         * @param value - The value to store with the given key.
+         * @param options - The Set options
+         * @return string or null If value isn't set because of `onlyIfExists` or `onlyIfDoesNotExist`
+         *     conditions, return null. If `returnOldValue` is set, return the old value as a string.
+         */
+        public TransactionBuilder set(String key, String value, SetOptions options) {
+            createOrAddCommand(Command.set(key, value, options.toArgs()));
+            return this;
+        }
+
+        private void createOrAddCommand(Command command) {
+            if (this.commands == null) {
```

Comment:
I don't this that this could ever happen.
probably remove `if` and rename the function

Reply:
Unfortunately, the default builder doesn't contain the linked list. 
It fails the first time we add values to `this.commands`.

Reply:
Nevermind - this has been removed.